src/sign-module.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+echo "Running the Axcient Kernel Module signer"
+
+result=$(mokutil --sb-state)
+if [[ "$result" != *"enabled"* ]]; then
+    echo "Detected a BIOS firmware or the Secure Boot feature is disabled"
+    exit 1
+else 
+    keys=$(cat /proc/keys | grep "asymmetri Axcient")
+    if [ -n "$keys" ]; then 
+	echo "Starting the setup"
+    else
+	echo "WARNING: Secure Boot detected on this system. Either disable EFI Secure Boot "
+	"or check with Axcient knowledgebase for instructions "
+	"for enrolling a signing key for secure DKMS module support." >&2 \
+	&& exit 0
+    fi	
+fi
+
+echo "Kernel Version: ${1}"
+echo "Module Path: ${2}"
+
+hash_algo=sha256
+private_key=/root/axcient-driver.key
```

Comment:
Whoose generate this keys?

Reply:
The use has to generate them. They have to do it manually unfortunately.